Draft of WASMEdge governance review. (#772)
cncf/tag-contributor-strategy: governance/assessments/projects/wasmedge/2025-02-12.md

# WASMEdge Governance Review

What follows is a governance review and assessment for the WASMEdge project. This review is carried out by members of the Governance Working Group of TAG Contributor Strategy. The review may have been done because of a change in maturity level for the project, at the request of the TOC, or as a request by the project itself. If requested by the project, the review will be provided to the project maintainers. Otherwise, the review will be submitted to the TOC for their follow-up.

Projects may ask TAG Contributor Strategy for assistance in resolving any issues uncovered by the review. The TAG is available via our [slack channel](https://cloud-native.slack.com/archives/CT6CWS1JN), [email](https://lists.cncf.io/g/cncf-tag-contributor-strategy), [GitHub](https://github.com/cncf/tag-contributor-strategy), or by joining our weekly meetings (listed on the [CNCF public calendar](https://www.cncf.io/calendar/)).

## Summary and Assessment

Status: Mostly Satisfactory

> WASMEdge has clearly made progress in all aspects of governance and contributor management since joining the CNCF.  They have many miscellaneous issues with governance and contributor documentation, which is normal for a project at this stage.  The project needs to use its advancement to Incubating in order to drive more contributor recruitment.

Reviewer: ```suggestion
WASMEdge has clearly made progress in all aspects of governance and contributor management since joining the CNCF. They have many miscellaneous issues with governance and contributor documentation, which is normal for a project at this stage.  The project needs to use its advancement to Incubating in order to drive more contributor recruitment.
```

Project documentation covers all required lifecycle areas for a small project.  See the [Governance Review request](https://github.com/cncf/tag-contributor-strategy/issues/690) for some examples by the project of management activities.

### Maintainers and Maintainer Lifecycle

* Document complete list of current maintainers, including names, contact information, domain of responsibility, and affiliation.
* A number of active maintainers which is appropriate to the size and scope of the project.

<!--- Check maintainers.md, governance.md, contributor ladder and other files.
Maintainers file should have contact info (not just GH handles), other useful information, and employer
Do a quick sanity check on the number of maintainers.  A very large critical project applying for graduation which has three maintainers is probably at bus-factor risk.  In contrast, a narrowly-scoped project with 50 maintainers is probably listing people who are not actually maintainers.
Also check that maintainers on the list are active using Devstats, GitHub and LFX Insights.  Query inactive maintainers. --->

> Currently WASMEdge has [four Maintainers](https://github.com/WasmEdge/WasmEdge/blob/master/docs/OWNER.md), which seems reasonable for a project this size.  They are also the foudning maintainers of WASMEdge, and there has yet to be any maintainer turnover.  All are active. The project has promoted many regular contributors to intermediate levels of code ownership (Reviewer and Committer), and seems prepared to eventually advance one or more contributors to maintainer according to the documented requiremens.

Reviewer: ```suggestion
Currently WASMEdge has [four Maintainers](https://github.com/WasmEdge/WasmEdge/blob/master/docs/OWNER.md), which seems reasonable for a project this size.  They are also the founding maintainers of WASMEdge, and there has yet to be any maintainer turnover.  All are active. The project has promoted many regular contributors to intermediate levels of code ownership (Reviewer and Committer), and seems prepared to eventually advance one or more contributors to maintainer according to the documented requirements.
```

The [Roadmap](https://github.com/WasmEdge/WasmEdge/blob/master/docs/ROADMAP.md) has a written process for adding items to it, and as far as I can tell the project has followed this public process for the last few quarters.

* Project must have, and document, at least one public communications channel for users and/or contributors.
* List and document all project communication channels, including subprojects (mail list/slack/etc.).  List any non-public communications channels and what their special purpose is.
* Up-to-date public meeting schedulers and/or integration with CNCF calendar.

The project has:

- [mailing list](https://groups.google.com/g/wasmedge/): rarely used
- [Discord](https://discord.com/invite/JHxMj9EQbA): very active

> - [community meeting](https://docs.google.com/document/d/1iFlVl7R97Lze4RDykzElJGDjjWYDlkI8Rhf8g4dQ5Rk/edit?tab=t.0: held regularly, videos uploaded

Reviewer: ```suggestion
- [community meeting](https://docs.google.com/document/d/1iFlVl7R97Lze4RDykzElJGDjjWYDlkI8Rhf8g4dQ5Rk/edit?tab=t.0:) held regularly, videos uploaded
```
